pkg/features/features.go

```diff
@@ -49,6 +49,13 @@ const (
 	// This gate enables Go runtime soft memory limit as explained in
 	// https://pkg.go.dev/runtime/debug#SetMemoryLimit
 	GoMemLimit featuregate.Feature = "GoMemLimit"
+
+	// PaginateAPICalls owner: @irfanurehman
+	// alpha:
+	//
+	// Pagination support for list API calls to API server querying workload controllers
+	// Without this feature gate the whole list is requested in a single list API call.
+	PaginateAPICalls featuregate.Feature = "PaginateAPICalls"
```

Comment:
Do you think if it makes sense to combine the `GoMemLimit` and `PaginateAPICalls` into one feature flag? I don't think it makes sense to enable one feature while disabling the other one. What do you think? I haven't thought of a good name though ...

Reply:
Made these into one flag `MemoryOptimisations=true`